src/analysis/modals/AnalysisModal.ts

```diff
@@ -254,22 +258,22 @@ export const AnalysisModal = withDisplayName('AnalysisModal')(
       hover: { backgroundColor: colors.accent(0.3) },
     };
 
-    const availableRuntimeTools = cloudRuntimeTools[cloudProvider];
-    const availableAppTools = cloudAppTools[cloudProvider];
+    const availableRuntimeTools: readonly AvailableRuntimeTool[] = cloudRuntimeTools[cloudProvider];
+    const availableAppTools: readonly AvailableAppTool[] = cloudAppTools[cloudProvider];
 
-    const currentApps: Record<AppToolLabel, App | undefined> = {
+    const currentApps: Record<AvailableAppToolLabel, App | undefined> = {
```

Comment:
Previously, these constants required an entry for _every_ app tool. However, they really only need an entry for the app tools in `cloudAppTools[cloudProvider]`. By declaring `cloudAppTools` as const, we can get a type that only includes the app tools that are in that constant.

https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-4.html#const-assertions